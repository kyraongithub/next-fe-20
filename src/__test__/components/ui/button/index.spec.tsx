import Button from '@/components/UI/button';
import { render, fireEvent } from '@testing-library/react';

describe('Button Component', () => {
  it('should render button', () => {
    const handleClick = jest.fn();
    const { container } = render(<Button onclick={handleClick}>Click</Button>);
    expect(container).toMatchSnapshot();
  });
  it('should click button', () => {
    const handleClick = jest.fn();
    const { getByText } = render(<Button onclick={handleClick}>Click</Button>);
    const button = getByText('Click');
    fireEvent.click(button);
    expect(handleClick).toHaveBeenCalledTimes(1);
  });
});
