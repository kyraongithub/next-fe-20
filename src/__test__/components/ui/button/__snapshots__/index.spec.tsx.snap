// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component should render button 1`] = `
<div>
  <button>
    Click
  </button>
</div>
`;
